### ISSUE TITLE
API support for updating node configuration online

### ISSUE THREAD

**Issue by cylewitruk** [stacks-network/stacks-blockchain#3361]
**Is your feature request related to a problem? Please describe.**
No

**Describe the solution you'd like**
It would be nice to have an API endpoint to update node configuration on-the-fly (online), specifically for miners, and more specifically the `satoshis_per_byte` and `burn_fee_cap` parameters.  Since (as I've understood it) there is a penalty of 6 blocks for miners who have just joined.  An API endpoint would be preferred over command-line commands so that it can be scripted (eventually from a separate machine/VM/container responsible for strategy calculations).

**Describe alternatives you've considered**
At the moment (again, if I've understood correctly) restarting a mining node will incur a penalty of 6 blocks.  While I don't see any practical alternatives at the moment (except somehow scripting an RBF for every transaction once the node has started..), another possible solution would be to allow nodes to disappear for [x amount of] time (minutes?), to allow for node updates/config updates without penalty.  This might actually be the fairest in the long run, as node updates should be encouraged and not penalized (maybe not necessarily related to this feature request).

**Additional context**
The documentation was slightly unclear when it comes to the `satoshis_per_byte` configuration parameter, and I was quite surprised when I was paying $7/commitment in just fees.  I then set this manually according to current bitcoin fees, but if fees were to rise I want to be able to control the node with my own logic, and not be penalized for restarting the node to read new configuration.

**Comment by jcnelson** [comment]
Hey, looks like there is some work-in-progress on this: https://github.com/stacks-network/stacks-blockchain/pull/3222

Want to give it a try?

**Comment by jcnelson** [comment]
(Sorry, fat-fingered the UI)

**Comment by cylewitruk** [comment]
That's one alternative, of course, and welcomed :)  However it does require having direct access to the filesystem to change the config file.  API endpoints would allow for an external process (e.g. a control process in another docker container) to react to external events and update the configuration values much more easily.

**Comment by jcnelson** [comment]
> However it does require having direct access to the filesystem to change the config file.

Yes, this is deliberate.  It separates the problem of updating the settings from the tool that carries out the updating.  It's entirely possible today to create an add-on admin service that would let you do this, without having to modify the node.

**Comment by cylewitruk** [comment]
> It's entirely possible today to create an add-on admin service that would let you do this, without having to modify the node.

I'm not disagreeing with you. However it would make things easier and generally more platform-agnostic.

Given the volatile nature of crypto, the ability to change settings on-the-fly and quickly (and likely by a custom-written control process) is important, particularly when needing to keep track of both bitcoin trx fees and the recent sats commitments of other miners. An API would be the easiest way to achieve this across all platforms.  

I personally can live with #3222, but I believe in Stacks and anything to help increase adopt-ability is therefore also important to me.

**Comment by Hero-Gamer** [comment]
Problem: Miner block commits seem to be missing some bitcoin blocks due to high bitcoin fees.
Do we need to revive this for better smarter more dynamic fee bidding for Stacks miners?

**Comment by stevenbooke** [comment]
@jcnelson Hello, I am new to the Stacks community and would like to contribute. I would like to clarify a couple things/get some pointers before I take a shot at this.

The feature request is an API/RPC that can update the node config while the node is running. From a high level perspective, could I achieve this by making the configuration be a shared state that gets updated when a request is made to this API/RPC?

Would the files I would be modifying most likely be `src/net/rpc.rs`, `src/net/https.rs`, and `src/net/mod.res`?

Any pointers on how to start would also be much appreciated! Thanks!
